Userland/Libraries/LibJS/Lexer.cpp

```diff
@@ -381,10 +396,14 @@ bool Lexer::is_identifier_start() const
     return false;
 }
 
-bool Lexer::is_identifier_middle() const
+bool Lexer::is_identifier_middle(size_t& identifier_length) const
 {
+    identifier_length = 1;
+
+    if (m_current_char == '\\')
+        return is_unicode_escape(identifier_length);
     if (!is_unicode_character())
-        return is_identifier_start() || is_ascii_digit(m_current_char);
+        return is_identifier_start(identifier_length) || is_ascii_digit(m_current_char);
```

Comment:
Could this be a problem with the call to `is_identifier_start`? 
Since that could allow characters which are in `ID_Start` but not in `ID_Continue`

Reply:
Actually it seems ID_Continue is defined to be a superset of `ID_Start` so this should not be a problem.